article/sections/_5_graph_testing.tex

```diff
@@ -1,14 +1,230 @@
+Here, we impute $Z$ by sampling from the prior or posterior distribution of $Z$, depending on whether we're testing independencies before or after performing inference on our model's parameters.
+Moreover, we'll now compute the p-value of this test statistic by marginalizing over the permutations and imputations.
+Specifically, our p-value will be
+\begin{equation}
+E_{\textrm{samples, permutations}} \left[ \mathbb{I} \left \lbrace R^2 \left( X, Y, Z_{\textrm{sampled}} \right) <  R^2 \left( X_{\textrm{sampled}}, Y_{\textrm{sampled}} ^{\textrm{permuted}}, Z_{\textrm{sampled}} \right) \right \rbrace \right]
+\end{equation}
+where $\mathbb{I}$ represents the indicator function that equals one if the condition inside its braces is true and zero otherwise.
+For reference, this is the same as the p-value for test statistics (or discrepancies) defined in \citet[Eq. 7]{gelman_1996_posterior}.
+
+\begin{figure}
+   \centering
+   \includegraphics[width=0.85\textwidth]{deconfounder-causal-graph}
+   \caption{Causal graph from applying the deconfounder algorithm \citep{wang_2019_blessings} to our dataset.}
+   \label{fig:deconfounder-graph}
+\end{figure}
+
+\begin{figure}
+   \centering
+   \includegraphics[width=0.85\textwidth]{latent-drivers-vs-num-autos}
+   \caption{Results of testing that the number of drivers is independent of the number of automobiles in the household, conditional on the latent variable.}
+   \label{fig:latent-cit-results}
+\end{figure}
+
+Lastly, because our test statistic is itself a random variable that depends on the imputed values of $Z$, we must change our visualization method.
+Instead of plotting a single line versus a distribution, we now plot two distributions against one another.
+We first plot the distribution of ``observed'' test statistics that we computed using the observed $X$, observed $Y$ and imputed $Z$ values.
+Then, we plot the distribution of ``sampled'' test statistics using prior samples of $\left( X, Y, Z \right)$ as a reference.
+Here, we have one value per imputed vector $Z$ in both the observed and sampled distributions.
+However, for the distribution of ``sampled'' test statistics, we marginalize over permutations of $Y$ since this distribution represents the null hypothesis of conditional or marginal independence.
+
+That last paragraph may have been confusing, so we'll walk through an example.
+Figure \ref{fig:deconfounder-graph} shows the causal graph implied by applying the deconfounder algorithm of \citet{wang_2019_blessings} to our dataset.
+We describe this application more thoroughly in Section \ref{sec:latent-confounding}.
+For now, we present the graph to highlight the assumptions that we will test.
+Specifically, we'll examine the assumption that the observed number of licensed drivers in a household is independent of the observed number of automobiles in that household, conditional on the latent variable $X^{*}$.
```

Comment:
Ah, I see now. So these tests only apply when you have a model to compute Z, e.g. the deconfounder algorithm.

So in other words, assuming that you *know* the latent variable Z, can we test whether X and Y are conditionally independent. 

This strikes me as a little bit unintuitive. Particularly, here are my thoughts/concerns:
- To re-frame the problem: we want to test conditional independence, conditional on a latent variable. To do that, we need some model of the latent confounder that gives us its posterior distribution, or, if we're testing conditional independence conditional on a prior distribution, we need to assume a prior. The test is then testing for many things at once, which to me makes it hard to interpret and I'm struggling to understand its usefulness. In particular, failing to reject the null can either mean i) our model of the confounder is wrong, or ii) there are additional confounders that are not represented. I noticed you touch on this at the end, but to me this is an important issue that makes it hard to understand the usefulness of this step. 
- A more specific comment on the illustration using the deconfounder: while I think this is a good illustration, I think we can frame it differently because of the goals of the deconfounder in the first place. In fact, the deconfounder algorithm is literally trying recover an vector X* that renders all variables conditionally independent -- so, failing to reject that the observed number of licensed drivers is independent of the observed number of automobiles conditional on X* is more a statement that the deconfounder did not do its job to recover a good X* than it is a statement on whether they are conditionally independent given X* (X* has no real meaning, it's just a vector that's supposed to make the other variables conditionally independent). Not sure if I'm making my point clearly but let's discuss more. 

In general, my suggestion is to change the objective and tone of this section entirely, and make it about testing the results of the deconfounder. I think that's more of a contribution in and by itself, but we should discuss more.

Reply:
I think we're mostly together.

Absolutely +1 on testing multiple things at once, and this multiplicity diminishing the value of the test. No free lunch!

In terms of the objective of the test being to verify the deconfounder, that makes sense for the version of the test using the posterior distribution!

For the version of the distribution that uses prior distributions, I think failing to reject is a statement that the observed data don't look like the simulated data (i.e., the prior sampled X*/Y*/Z*, where X* and Y* are independent of Z*). The real usefulness of such a test is when you've created a prior such that \int P(X*|Z*)P(Z*) [dZ*] is a reasonable marginal prior for P(X) and when \int P( Y* | Z* ) P( Z* ) [dZ*]  is a reasonable prior for P(Y). At this point, the remaining discrepancies between observed and simulated are due to the independence assumption being violated.

I added two sentences to highlight this!